_invokers/java.md

```diff
@@ -9,46 +9,180 @@ href: https://github.com/projectriff/java-function-invoker
+It will then invoke the function for each request.
+When used in a function service like [riff on Knative](https://projectriff.io/), the invoker boot application is provided by the platform when functions are built.
+
+## How To Use It
+
+### Function source
+
+You need to configure a Maven or Gradle project for your function.
+If you use Spring Boot then we recommend using [Spring Initializr](https://start.spring.io/) to bootstrap your project.
+If you are not using Spring Boot for your function code then you need to create your own build configuration.
+There are no required dependencies from the Java Function Invoker, it only requires that your function implements the `java.util.function.Function` interface.
+
+The [Spring Cloud Function](https://cloud.spring.io/spring-cloud-function/) project provides support for writing functions as part of a Spring Boot app.
+The Java Function Invoker uses Spring Cloud Function to invoke your functions, so most features available with Spring Cloud Function are also available with the Java Function Invoker.
+
+Example of a plain Java function:
+
+```java
+package functions;
+
+import java.util.function.Function;
+
+public class Upper implements Function<String, String> {
+
+    public String apply(String name) {
+        return name.toUpperCase();
+    }
+}
+```
+
+Example of a Spring Boot app with a function bean:
+
+```java
+package com.example.uppercase;
+
+import java.util.function.Function;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+public class UppercaseApplication {
+
+	@Bean
+	Function<String, String> uppercase() {
+		return s -> s.toUpperCase();
+	}
+
+	public static void main(String[] args) {
+		SpringApplication.run(UppercaseApplication.class, args);
+	}
+}
+```
+
+#### Function handler
+
+The simplest form of the handler is a bean name or a class name that can be instantiated (with a default constructor).
+More complex creation scenarios can be handled by giving the handler in the form `<bean>[&main=<main>]` where
 
 * `<bean>` is a class name or bean name, and
 * `<main` is a Spring `@Configuration` class to create an application context
 
-If you provide a `main` parameter you need to include Spring Boot and
-all the other dependencies of the context in your archive. A Spring
-Boot jar file does not need an explicit `main` (there is one in the
-`MANIFEST.MF`), but if you supply one it will be used. If the
-`Function` has POJO (i.e. not generic JDK classes) as parameter types,
-then you also need to depend on `spring-cloud-function-context` (and
-include that in the archive).
+If you provide a `main` parameter you need to include Spring Boot and all the other dependencies of the context in your archive.
+A Spring Boot jar file does not need an explicit `main` (there is one in the `MANIFEST.MF`), but if you supply one it will be used.
```

Comment:
I don't think this is true, is it? cc @dsyer

Reply:
It is true. That's what a "main" is (a Spring configuration class). If you don't have Spring in your app you don't need it.